Source/mpqapi.cpp

```diff
@@ -3,349 +3,44 @@
  *

```

Comment:
Looks like the unmodified Devilution version of mpqapi.cpp was used for this port (likely to save time on porting DevilutionX's C++11 to C++03). This is likely one of the major obstacles to merging the Xbox code. I have hacked together a cross-platform version from an older version of DevilutionX before we did a major rework on this code (afaik this was needed for stable Amiga support), this should be good enough to get us 90% of the way before we have to address it.

Reply:
Yep was taken from unmodified Devilution version. I have made the changes required for Xbox to build your cross-platform version.

Below are the changes to get the latest head revision building on Xbox, tested and working well: https://github.com/marty28/devilutionX/commit/39ef7de1c85af787c25c4b65dc8f78fcfde390ea

Cheers